app/components/Settings/Scheduler/previewModal.jsx

```diff
@@ -0,0 +1,248 @@
+    }
+
+    onSubmit() {
+        if (this.state.editing) {
+            const { connectionId, fid, requestor, uids } = this.props.query;
+            const { code: query, refreshInterval } = this.state;
+            this.props.onSave({
+                connectionId,
+                fid,
+                requestor,
+                uids,
+                query,
+                refreshInterval
+            });
+            this.setState({ editing: false, confirmedDelete: false });
+        } else {
+            this.toggleEditing();
+        }
+    }
+
+    onDelete() {
+        if (this.state.confirmedDelete) {
+            this.setState({ confirmedDelete: false }, () => {
+                this.props.onDelete(this.props.query.fid);
+            });
+        } else {
+            this.setState({ confirmedDelete: true });
+        }
+    }
+
+    close() {
+        this.setState({ confirmedDelete: false });
+        this.props.onClickAway();
+    }
+
+    render() {
+        const props = this.props;
+
+        let content;
+        if (!props.query) {
+            content = null;
+        } else {
+            const [account, gridId] = props.query.fid.split(':');
+            const link = `${plotlyUrl()}/~${account}/${gridId}`;
+            const editing = this.state.editing;
+            content = (
+                <Column style={{ width: '50%', background: 'white' }}>
+                    <Row
+                        className="sql-preview"
+                        style={{
+                            padding: '32px',
+                            position: 'relative',
+                            justifyContent: 'flex-start'
+                        }}
+                    >
+                        <h5
+                            className="sql-preview"
+                            style={{ margin: 0, letterSpacing: '1px' }}
+                        >
+                            <SQL className="bold">{props.query.query}</SQL>
+                        </h5>
+                        <button
+                            onClick={this.close}
+                            style={{
+                                position: 'absolute',
+                                top: '16px',
+                                right: '16px',
+                                padding: '2px 4px'
+                            }}
+                        >
+                            &times;
+                        </button>
+                    </Row>
+                    <Column style={{ background: '#F5F7FB', padding: '32px' }}>
+                        <Row style={rowStyle}>
+                            <div style={boxStyle}>Query</div>
+                            <div
+                                className="sql-preview scheduler"
+                                style={boxStyle}
+                            >
+                                {editing ? (
+                                    <div>
+                                        <CodeMirror
```

Comment:
There is an issue with size of the cursor. See below:

![peek 2018-07-04 15-51](https://user-images.githubusercontent.com/6199391/42283707-27e1159e-7fa2-11e8-9e96-80fd41903ec3.gif)